extension/lib/shopgate/Authorisation.js

```diff
@@ -0,0 +1,49 @@
+const ShopgateExtensionStorage = require('./ExtensionStorage')
+const shopifyAdminClient = require('../shopify/AdminClient')
+
+const STOREFRONT_AUTH_TOKEN = 'storefront-auth-token'
+
+class ShopgateAuthorisation {
+  /**
+   * @param {ShopgateExtensionStorage} pipelineStorage
+   * @param {ShopifyAdminClient} shopifyAdminClient
+   */
+  constructor (pipelineStorage, shopifyAdminClient) {
+    this._pipelineStorage = pipelineStorage
+    this._shopifyAdminClient = shopifyAdminClient
+  }
+
+  /**
+   * @returns {Promise<string>}
+   */
+  async acquireShopifyStorefrontToken () {
+    let shopifyStorefrontAccessToken = await this._pipelineStorage.get(STOREFRONT_AUTH_TOKEN)
```

Comment:
can this be const?

Reply:
It's value is overwritten few lines later. In general it could be written differently in order to have const - with an early return for example - is this what you mean?

Reply:
overlooked its replacement. Thats fine